pkg/operator/controllers/gameserverbuild_controller.go

```diff
@@ -142,27 +142,35 @@ func (r *GameServerBuildReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 		if gs.Status.State == "" {
 			pendingCount++
-		} else if gs.Status.State == mpsv1alpha1.GameServerStateInitializing {
+		} else if gs.Status.State == mpsv1alpha1.GameServerStateInitializing && gs.Status.Health == mpsv1alpha1.GameServerHealthy {
 			initializingCount++
-		} else if gs.Status.State == mpsv1alpha1.GameServerStateStandingBy {
+		} else if gs.Status.State == mpsv1alpha1.GameServerStateStandingBy && gs.Status.Health == mpsv1alpha1.GameServerHealthy {
 			standingByCount++
-		} else if gs.Status.State == mpsv1alpha1.GameServerStateActive {
+		} else if gs.Status.State == mpsv1alpha1.GameServerStateActive && gs.Status.Health == mpsv1alpha1.GameServerHealthy {
 			activeCount++
+		} else if gs.Status.State == mpsv1alpha1.GameServerStateGameCompleted && gs.Status.Health == mpsv1alpha1.GameServerHealthy {
```

Comment:
Just a question, this case is for when the GameServer finished the session and exited gracefully?

Reply:
yup and I think that this is the case when Kubernetes evicts Pods, so they get an exit code of zero so they transition to the Completed state